pants_plugins/sendwave/pants_docker/package.py

```diff
@@ -114,24 +110,26 @@ async def package_into_image(
         if component.sources:
             source_digests.append(component.sources)
         run_commands.extend(component.commands)
-    source_digest = await Get(Digest, MergeDigests(digests=source_digests))
-    application_digest = await Get(Digest, AddPrefix(source_digest, "application"))
-    snapshot = await Get(Snapshot, Digest, application_digest)
-    logger.info(snapshot.files)
+    source_digest = await Get(Digest, MergeDigests(source_digests))
+    application_snapshot = await Get(Snapshot, AddPrefix(source_digest, "application"))
+    logger.info(application_snapshot.files)
+
     dockerfile_contents = _create_dockerfile(
         field_set.base_image.value,
-        field_set.workdir and field_set.workdir.value,
+        field_set.workdir.value,
```

Comment:
I'm not sure what this was after?

Reply:
this was just a way to check if it was set - since it will return a falsy value if the field is missing(?) open to a better way of doing this

Reply:
The field will always be defined, only the value might be `None`. Your type hint already accounts for `Optional[str]`.